agrirouter-middleware-controller/src/main/java/de/agrirouter/middleware/controller/secured/EndpointController.java

```diff
@@ -103,4 +100,66 @@ public EndpointController(ApplicationService applicationService,
         return ResponseEntity.ok(new EndpointStatusResponse(mappedEndpoints));
     }
 
+    /**
+     * Fetch the health status for an endpoint. This is the more common way to have a quick health check for the endpoint.
+     *
+     * @param externalEndpointId -
+     * @return -
+     */
+    @GetMapping(
+            value = "/{externalEndpointId}"
+    )
+    @Operation(
+            operationId = "endpoint.status.health",
+            description = "Fetch the health status of an existing endpoint.",
+            responses = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "The endpoint is connected and is able to communicate."
+                    ),
+                    @ApiResponse(
+                            responseCode = "404",
+                            description = "In case that the endpoint was not found."
+                    ),
+                    @ApiResponse(
+                            responseCode = "400",
+                            description = "In case of an business exception.",
+                            content = @Content(
+                                    schema = @Schema(
+                                            implementation = ErrorResponse.class
+                                    ),
+                                    mediaType = MediaType.APPLICATION_JSON_VALUE
+                            )
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "In case of an unknown error.",
+                            content = @Content(
+                                    schema = @Schema(
+                                            implementation = ErrorResponse.class
+                                    ),
+                                    mediaType = MediaType.APPLICATION_JSON_VALUE
+                            )
+                    ),
+                    @ApiResponse(
+                            responseCode = "503",
+                            description = "In case the endpoint is currently not connected."
+                    )
+            }
+    )
+    public ResponseEntity<Void> healthStatus(@Parameter(description = "The external endpoint id.", required = true) @PathVariable String externalEndpointId) {
```

Comment:
Not sure if there is a particular reason for doing it this way, but this seems like its a great fit for "Custom actuator endpoints", see [here](https://www.baeldung.com/spring-boot-actuators#custom-endpoint).

An argument for doing it this way would be of course that its automatically documented in the swagger contract.

Reply:
This is right, this could be part of the actuator (and will be). But the main reason is, that the custom route should be part of the documentation. And the domain logic is kept close to the controller.